pkg/flow/flowtest/flowtest_test.go

```diff
@@ -0,0 +1,22 @@
+package flowtest_test
+
+import (
+	"io/fs"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/agent/pkg/flow/flowtest"
+	"github.com/stretchr/testify/require"
+)
+
+func Test(t *testing.T) {
+	filepath.WalkDir("testdata/", func(path string, d fs.DirEntry, err error) error {
+		if filepath.Ext(path) != ".txtar" {
+			return nil
+		}
+		t.Run(d.Name(), func(t *testing.T) {
+			require.NoError(t, flowtest.TestScript(path))
```

Comment:
Could we instead pass the `testing.T` to `TestScript` and further, so that we can use the rich assert/require library? These can facilitate stuff like comparing maps, checking numbers are within a range, checking something is contained in the other thing, using the 'eventually' functions... the error messages produced by these are usually quite clear and useful. So I think it's worth to make it possible to use them - especially will be useful for all the asserting commands.